model/github_app_auth.go

```diff
@@ -42,16 +43,80 @@ func githubAppCheckAndRunParameterStoreOp(ctx context.Context, appAuth *githubap
 	op(ref, isRepoRef)
 }
 
-// githubAppAuthUpsert upserts the GitHub app auth into the database and to
-// Parameter Store if enabled.
+// GitHubAppAuthFindOne finds the GitHub app auth and retrieves the private key
+// from Parameter Store if enabled.
+func GitHubAppAuthFindOne(id string) (*githubapp.GithubAppAuth, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultParameterStoreAccessTimeout)
+	defer cancel()
+
+	appAuth, err := githubapp.FindOneGithubAppAuth(id)
+	if err != nil {
+		return nil, err
+	}
+	if appAuth == nil {
+		return nil, nil
+	}
+
+	githubAppCheckAndRunParameterStoreOp(ctx, appAuth, func(ref *ProjectRef, isRepoRef bool) {
+		if ref.ParameterStoreGitHubAppSynced {
+			if err := githubAppAuthFindParameterStore(ctx, appAuth); err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
+					"message":    "could not find GitHub app private key in Parameter Store",
+					"op":         "FindOne",
+					"project_id": appAuth.Id,
+					"epic":       "DEVPROD-5552",
+				}))
+			}
+		}
+	}, "FindOne")
+
+	return appAuth, nil
+}
+
+func githubAppAuthFindParameterStore(ctx context.Context, appAuth *githubapp.GithubAppAuth) error {
+	paramMgr := evergreen.GetEnvironment().ParameterManager()
+
+	params, err := paramMgr.GetStrict(ctx, appAuth.PrivateKeyParameter)
+	if err != nil {
+		return errors.Wrapf(err, "getting GitHub app private key from Parameter Store")
+	}
+
+	privKey := []byte(params[0].Value)
```

Comment:
Preference to either error or return nil on a nil slice as an extra precaution

Reply:
`GetStrict` already checks that each requested name returns a corresponding result, but it doesn't hurt to check again.